python/cudf/cudf/core/column/struct.py

```diff
@@ -80,6 +80,15 @@ def to_arrow(self):
             pa_type, len(self), buffers, children=children
         )
 
+    def __getitem__(self, args):
+        result = super().__getitem__(args)
+        if isinstance(result, dict):
+            return dict(
+                (field, value)
+                for field, value in zip(self.dtype.fields, result.values())
+            )
```

Comment:
dict comprehension is just a little cleaner here:

```suggestion
            return {
                field: value for field, value in zip(self.dtype.fields, result.values())
            }
```

Reply:
May fail style check so don't accept the suggestion directly :)